src/main/resources/static/javascript/homepage.js

```diff
@@ -1,36 +1,35 @@
 $(document).ready(function () {
 
+    var apiKey = "AIzaSyCYf-bigRT20CHeW79bBbnBjtp0RPeTfes"; //restricted api key
```

Comment:
shouldn't this be treated as a secret?

Reply:
This is used inside javascript and will be rendered to the client-side so that it cannot be hidden. 
GCP provides restrictions about the usage of the API key, only certain IP addresses can invoke this API key.